packages/application-shell/src/components/user-settings-menu/user-settings-menu.tsx

```diff
@@ -233,36 +233,49 @@ const UserSettingsMenuBody = (props: MenuBodyProps) => {
 };
 UserSettingsMenuBody.displayName = 'UserSettingsMenuBody';
 
-const UserSettingsMenu = (props: Props) => (
-  <div data-test="user-settings-menu">
-    <Downshift stateReducer={stateReducer}>
-      {(downshiftProps) => (
-        <div>
-          <button
-            role="user-menu-toggle"
-            css={css`
-              cursor: pointer;
-              border: none;
-              padding: 0;
-              display: flex;
-              background: transparent;
-            `}
-            {...downshiftProps.getToggleButtonProps()}
-          >
-            <UserAvatar
-              firstName={props.firstName}
-              lastName={props.lastName}
-              gravatarHash={props.gravatarHash}
-            />
-          </button>
-          {downshiftProps.isOpen && (
-            <UserSettingsMenuBody {...props} downshiftProps={downshiftProps} />
-          )}
-        </div>
-      )}
-    </Downshift>
-  </div>
-);
+const UserSettingsMenu = (props: Props) => {
+  const intl = useIntl();
+
+  return (
+    <div data-test="user-settings-menu">
+      <Downshift stateReducer={stateReducer}>
+        {(downshiftProps) => (
+          <div>
+            <button
+              role="user-menu-toggle"
+              css={css`
+                cursor: pointer;
+                border: none;
+                padding: 0;
+                display: flex;
+                background: transparent;
+              `}
+              {...downshiftProps.getToggleButtonProps({
+                'aria-label': intl.formatMessage(
+                  downshiftProps.isOpen
+                    ? messages.closeMenuLabel
+                    : messages.openMenuLabel
```

Comment:
This is the relevant change. If the menu is open we show a label to close it and if closed one to open it.